### PR TITLE
refactor(ivy): migrate ɵɵ prefix back to Δ

### PR DESCRIPTION
**Pull request by benlesh**
Now that issues are resolved with Closure compiler, we can move back to our desired prefix of `Δ`.

### CODE REVIEW

File: aio/tools/transforms/angular-api-package/index.js [hunk 1]
```diff
@@ -9,197 +9,209 @@ const Package = require('dgeni').Package;
...
+            readTypeScriptModules, readFilesProcessor, collectExamples, tsParser,
+            packageContentFileReader) {
+
+          // Tell TypeScript how to load modules that start with with `@angular`
+          tsParser.options.paths = {'@angular/*': [API_SOURCE_PATH + '/*']};
+          tsParser.options.baseUrl = '.';
+
+          // API files are typescript
+          readTypeScriptModules.basePath = API_SOURCE_PATH;
+          readTypeScriptModules.ignoreExportsMatching = [/^_|^Δ|^VERSION$/];
```
Comment (mhevery) [inline aio/tools/transforms/angular-api-package/index.js]: can you not reformat this file?

**Review by mhevery — APPROVED** [review]

**Comment by benlesh** [comment]
[presubmit](http://test/OCL:248079577:BASE:248079593:1557814765150:db471ce5)

**Comment by gkalpak** [comment]
The reason the `test_aio_local`/`test_aio_local_ivy` jobs (which use the locally built Angular packages) are failing is Material components are trying to import things like `ɵɵdefineInjectable` and `ɵɵinject` from `@angular/core`, which no longer exist (for example see [material/esm2015/icons.js][1]).

There are warning about this in the `yarn build-local` step on the [CircleCI job][2].
No idea why these are just warnings and not hard failures 😕 (It does fail at runtime.)

##
So, if there are published packages that use `ɵɵ`-prefixed exports, what is the plan for them with the switch to `Δ`?
Or is it just material?

[1]: https://unpkg.com/@angular/material@8.0.0-rc.1/esm2015/icon.js
[2]: https://circleci.com/gh/angular/angular/318364

**Comment by ocombe** [comment]
Why is material using private APIs? The point of these being private is that we should be able to change them without any issue

**Review by ocombe — APPROVED** [review]

**Comment by devversion** [comment]
@ocombe Material is not using these directly. Since Angular Material uses tree-shakeable injectables which were introduced in V6 ([see here](https://github.com/angular/angular/commit/235a235)), NGC generates `ngInjectableDef` static class members which rely on the `defineInjectable` function were generated.

See for example: https://github.com/angular/material2-builds/blob/master/bundles/material-core.umd.js

Ideally the `defineInjectable` public symbol would have never been prefixed/updated in such a way as it breaks backwards compatibility.

**Comment by benlesh** [comment]
Ugh... I think they were supposed to use `defineInjectable` which is marked as a `@deprecated` `@publicApi` and not `@codeGenApi` like `ɵɵdefineInjectable`. :man_facepalming: 

Not sure what to do here. I don't think we want TWO deprecated versions of `defineInjectable`. 

@misko?

**Comment by gkalpak** [comment]
Who is "they"? According to https://github.com/angular/angular/pull/30362#issuecomment-492259806, this is `ngc`'s doing (which would imply other 3rd party libraries are affected?).

**Comment by devoto13** [comment]
I'm currently updating a library to compile it with Angular 8 and I believe I face a related issue. `ngc` from Angular 6 and 7 was emitting:

`FaIconService.ngInjectableDef = i0.defineInjectable({ factory: function FaIconService_Factory() { return new FaIconService(); }, token: FaIconService, providedIn: "root" });`

While from Angular 8 it emits: 

`FaIconService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function FaIconService_Factory() { return new FaIconService(); }, token: FaIconService, providedIn: "root" });`

As far as I can see updating library to Angular 8 forces it to drop support for all previous versions. Just want to double check that this change was intentional?

**Comment by benlesh** [comment]
Okay... so I'm just deprecating `ɵɵdefineInjectable` and `ɵɵinject` until we're out of RC, at which point it will be deleted.

**Comment by benlesh** [comment]
@devoto13 The angular team is dedicated to backward compatibility. This is probably just a growing pain with Ivy and the RC. Please file an issue to let us know what problems your experiencing, ideally with a repro. Then you can DM it to me and I'll try to get it escalated so we discuss it. We definitely don't want to break anyone  :)

(To DM me, the best bet is [twitter](https://twitter.com/benlesh), my GitHub notifications are a trainwreck)

**Comment by benlesh** [comment]
[newer presubmit](http://test/OCL:248079577:BASE:248181368:1557859526950:4f5c9438)

**Review by alxhub — APPROVED** [review]

**Review by IgorMinar — APPROVED** [review]
LGTM but please write more descriptive commit messages explaining why we are making this change at all.

**Comment by IgorMinar** [comment]
note from multiple 1:1 discussions with @benlesh, @alxhub, @mhevery and @kara: @benlesh is to create a new PR targeting the 8.0.x branch so that we can keep on cherry-picking fixes and infrastructure improvements between the two branches easily.

**Pull request closed without merging**